job.go

```diff
@@ -60,6 +61,13 @@ func (j *internalJob) stop() {
 	j.cancel()
 }
 
+func (j *internalJob) stopTimeReached() bool {
```

Comment:
In order to support time, mocking, this function should accept a now time.Time value. then, when it is called, we should pass in the schedulers clockwork time now value